### DISCUSSION TITLE
Wallet for customers using giftcards

### DISCUSSION THREAD

**Discussion by jeriox** [pretix/pretix#3934] [Ideas]
We have a legacy system that keeps a wallet for every customer, which they can top up and pay with at various points, e.g. coffee machines and our pretix shop. As the backend code is hard to maintain, we are currently in the process of porting this over to a pretix plugin. While the API part to external sales channels is very custom to us, the general idea of having a wallet for every customer seems like it could also work for pretix core. If you are interested, take a look at https://github.com/fsr-de/pretix-wallet/tree/2fc7a558890dfc1aeba1d930f9ea02d7ed30e923 where I implemented this based on customer accounts and gift cards. It is a rough first version, but already works fine.

**Comment by raphaelm** [comment]
Thanks! We do plan to add some concept of a connection between customer accounts and gift cards as a core feature in the future!
